### PR TITLE
Fix clippy warnings and errors

### PR DESCRIPTION
**Pull request by tesuji**
Clippy log here: https://travis-ci.com/lzutao/rustup.rs/jobs/194219923#L573

**Note**: I categorized each kind of clippy warnings to each commit. So it would be 
easier to review if you read commit one-by-one rather than read it all
in `File changed` tab.

### CODE REVIEW

File: download/tests/download-curl-resume.rs [hunk 1]
```diff
@@ -47,9 +48,8 @@ fn callback_gets_all_data_as_if_the_download_happened_all_at_once() {
         Some(&|msg| {
             match msg {
                 Event::ResumingPartialDownload => {
-                    let mut flag = callback_partial.lock().unwrap();
-                    assert!(!*flag);
-                    *flag = true;
+                    assert!(!callback_partial.load(Ordering::SeqCst));
+                    callback_partial.store(true, Ordering::SeqCst);
```
Comment (kinnison) [inline download/tests/download-curl-resume.rs:52]: Does this have the same side-effect of locking callback_partial for the duration of the read/assert/set sequence?

File: download/tests/download-reqwest-resume.rs [hunk 2]
```diff
@@ -47,9 +48,8 @@ fn callback_gets_all_data_as_if_the_download_happened_all_at_once() {
         Some(&|msg| {
             match msg {
                 Event::ResumingPartialDownload => {
-                    let mut flag = callback_partial.lock().unwrap();
-                    assert!(!*flag);
-                    *flag = true;
+                    assert!(!callback_partial.load(Ordering::SeqCst));
+                    callback_partial.store(true, Ordering::SeqCst);
```
Comment (kinnison) [inline download/tests/download-reqwest-resume.rs:52]: Ditto, locking semantics - do they change?

**Review by kinnison — COMMENTED** [review]
Apart from my one (probably irrelevant) concern about lock semantics in the mutex->atomicbool change, I'm happy

File: download/tests/download-curl-resume.rs (hunk 1 above)
Comment (tesuji) [reply download/tests/download-curl-resume.rs:52]: [I think so](https://doc.rust-lang.org/nightly/std/sync/atomic/enum.Ordering.html#variant.SeqCst).

**Pull request merged**